src/components/app/data/hooks/useEnterpriseLearner.test.jsx

```diff
@@ -19,37 +20,92 @@ const mockEnterpriseCustomer = enterpriseCustomerFactory();
 const mockAuthenticatedUser = authenticatedUserFactory();
 const mockEnterpriseLearnerData = {
   enterpriseCustomer: mockEnterpriseCustomer,
-  enterpriseCustomerUserRoleAssignments: [],
   activeEnterpriseCustomer: null,
   activeEnterpriseCustomerUserRoleAssignments: [],
   allLinkedEnterpriseCustomerUsers: [],
   enterpriseFeatures: {},
   staffEnterpriseCustomer: null,
+  shouldUpdateActiveEnterpriseCustomerUser: false,
 };
 
+const mockBFFDashboardData = {
+  enterpriseCustomer: {
+    ...mockEnterpriseCustomer,
+    isBFFEnabled: true,
+  },
+  allLinkedEnterpriseCustomerUsers: [],
+  enterpriseFeatures: {
+    isBFFEnabled: true,
+  },
+  shouldUpdateActiveEnterpriseCustomerUser: false,
+  enterpriseCustomerUserSubsidies: {
+    subscriptions: {
+      customerAgreement: {},
+      subscriptionLicenses: [],
+      subscriptionLicensesByStatus: {
+        activated: [],
+        assigned: [],
+        expired: [],
+        revoked: [],
+      },
+    },
+  },
+  enterpriseCourseEnrollments: [],
+  errors: [],
+  warnings: [],
+};
+
+const mockExpectedEnterpriseLearner = (isMatchedRoute) => (isMatchedRoute
+  ? {
+    enterpriseCustomer: mockBFFDashboardData.enterpriseCustomer,
+    allLinkedEnterpriseCustomerUsers: mockBFFDashboardData.allLinkedEnterpriseCustomerUsers,
+    enterpriseFeatures: mockBFFDashboardData.enterpriseFeatures,
+  }
+  : mockEnterpriseLearnerData);
+
 describe('useEnterpriseLearner', () => {
-  const Wrapper = ({ children }) => (
+  const Wrapper = ({ routes = null, children }) => (
     <QueryClientProvider client={queryClient()}>
-      <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
-        {children}
-      </AppContext.Provider>
+      <MemoryRouter initialEntries={[routes]}>
+        <AppContext.Provider value={{ authenticatedUser: mockAuthenticatedUser }}>
+          {children}
+        </AppContext.Provider>
+      </MemoryRouter>
     </QueryClientProvider>
   );
   beforeEach(() => {
     jest.clearAllMocks();
     fetchEnterpriseLearnerData.mockResolvedValue(mockEnterpriseLearnerData);
+    fetchEnterpriseLearnerDashboard.mockResolvedValue(mockBFFDashboardData);
     useParams.mockReturnValue({ enterpriseSlug: mockEnterpriseCustomer.slug });
   });
-  it('should handle resolved value correctly', async () => {
-    const { result, waitForNextUpdate } = renderHook(() => useEnterpriseLearner(), { wrapper: Wrapper });
+  it.each([
+    { isMatchedRoute: false },
+    { isMatchedRoute: true },
+  ])('should return enterprise learners correctly (%s)', async ({ isMatchedRoute }) => {
+    const mockSelect = jest.fn(data => data.transformed);
+    const { result, waitForNextUpdate } = renderHook(
+      () => {
+        if (isMatchedRoute) {
+          return useEnterpriseLearner({ select: mockSelect });
+        }
+        return useEnterpriseLearner();
+      },
```

Comment:
Similar comments as other tests around:
* nit: perhaps `isBFFMatchedRoute` or otherwise to be slightly more explicit
* [clarification] Do we need to test custom `select` function when `isMatchedRoute=false`?